src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -149,10 +149,14 @@ ucs_status_t uct_cuda_copy_ep_get_short(uct_ep_h tl_ep, void *buffer,
                                         unsigned length, uint64_t remote_addr,
                                         uct_rkey_t rkey)
 {
+    uct_cuda_copy_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_cuda_copy_iface_t);
     ucs_status_t status;
 
-    status = UCT_CUDA_FUNC(cudaMemcpy(buffer, (void *)remote_addr,
-                                      length, cudaMemcpyDeviceToHost));
+    UCT_CUDA_COPY_CHECK_AND_CREATE_STREAM(iface->stream_d2h);
+
+    UCT_CUDA_FUNC(cudaMemcpyAsync((void *)remote_addr, buffer, length,
+                                  cudaMemcpyHostToDevice, iface->stream_d2h));
```

Comment:
@Akshay-Venkatesh used wrong direction (cudaMemcpyHostToDevice) and src, dst parms

Reply:
shoot! Fixing this now